### PR TITLE
reverse_complement: Pre-allocate a buffer based on input size

### LINKED ISSUE #43: reverse_complement spends most of its time reading input
On my system, `reverse_complement < input25000000.txt` spends 67% of its time on the `read_to_end` call here.  According to the profile, that time is almost all spent in `memmove`.

```rust
    let mut data = Vec::with_capacity(1024 * 1024);
    stdin.lock().read_to_end(&mut data).unwrap();
```

Increasing the initial buffer size to fit the entire dataset cuts the `read_to_end` time in half, and reduces the total execution time by about 30%, but obviously it also wastes memory if the data is small.  (Since this program reads from stdin, it doesn't known the total size before it starts reading.)

### PR DESCRIPTION
**Pull request by mbrubeck**
Fixes #43.  This speeds up `reverse_complement < input25000000.txt` by about 40% on my system.

### CODE REVIEW

**Review by TeXitoi — APPROVED** [review]

**Comment by TeXitoi** [comment]
Great work! Please add your name in the contributors.

File: src/reverse_complement.rs [hunk 1]
```diff
@@ -187,10 +189,18 @@ fn parallel<'a, I, T, F>(iter: I, ref f: F)
     for jh in jhs { jh.join().unwrap(); }
 }
 
+fn file_size(f: &mut File) -> io::Result<usize> {
+    let cur = f.seek(SeekFrom::Current(0))?;
+    let end = f.seek(SeekFrom::End(0))?;
+    f.seek(SeekFrom::Start(cur))?;
+    Ok(end as usize + 1)
```
Comment (TeXitoi) [inline src/reverse_complement.rs]: Does `f.metadata()?.len()` works?

**Comment by mbrubeck** [comment]
> Does `f.metadata()?.len()` works?

Yes, it does. Thanks! Fixed to use `metadata()` and added my name.

**Review by TeXitoi — APPROVED** [review]

**Comment by TeXitoi** [comment]
https://alioth.debian.org/tracker/index.php?func=detail&aid=315633&group_id=100815&atid=413122 Thanks again for these contributions! Waiting the next ones ;-)

**Pull request merged**